scripts/portconfig

```diff
@@ -126,6 +126,13 @@ class portconfig(object):
     def set_fec(self, port, fec):
         if self.verbose:
             print("Setting fec %s on port %s" % (fec, port))
+        supported_fecs = self.get_supported_fecs(port)
+        if fec not in supported_fecs:
```

Comment:
In case `STATE_DB|<port>.supported_fecs` is `N/A`, all the defined FEC modes will be put into the variable `supported_fecs` here. From the user's PoV, the old logic is followed.
Please check [get_supported_fecs](https://github.com/Azure/sonic-utilities/pull/2223/files/0b18733145d342ee31921fdb6630c0b9339a1032#diff-141a107b5364385777ab15580e5797c054236c37780039c007621f507e449ba5R223) for more detail.